src/prefect/infrastructure/provisioners/container_instance.py

```diff
@@ -0,0 +1,765 @@
+            f"--location {self._location} --subscription {self._subscription_id} "
+            "--restart-policy OnFailure --output json"
+        )
+        await self.azure_cli.run_command(
+            create_command,
+            success_message=(
+                f"Container instance '{container_name}' created successfully in"
+                f" resource group '{self.RESOURCE_GROUP_NAME}' in location"
+                f" '{self._location}' in subscription '{self._subscription_name}'"
+            ),
+            failure_message=(
+                f"Failed to create container instance '{container_name}' in resource"
+                f" group '{self.RESOURCE_GROUP_NAME}' in location '{self._location}' in"
+                f" subscription '{self._subscription_name}'"
+            ),
+            ignore_if_exists=True,
+        )
+        return
+
+    async def _create_aci_credentials_block(
+        self,
+        work_pool_name: str,
+        client_id: str,
+        tenant_id: str,
+        client_secret: str,
+        client: PrefectClient,
+    ) -> UUID:
+        """
+        Creates a credentials block for Azure Container Instance.
+
+        Args:
+            work_pool_name (str): The name of the work pool.
+            client_id (str): The client ID obtained from app registration.
+            tenant_id (str): The tenant ID obtained from the secret generation.
+            client_secret (str): The client secret obtained from the secret generation.
+            client (PrefectClient): An instance of PrefectClient.
+
+        Returns:
+            UUID: The ID of the created credentials block.
+
+        Raises:
+            ObjectAlreadyExists: If a credentials block with the same name already exists.
+        """
+        credentials_block_name = f"{work_pool_name}-push-pool-credentials"
+        credentials_block_type = await client.read_block_type_by_slug(
+            "azure-container-instance-credentials"
+        )
+
+        credentials_block_schema = (
+            await client.get_most_recent_block_schema_for_block_type(
+                block_type_id=credentials_block_type.id
+            )
+        )
+
+        try:
+            block_doc = await client.create_block_document(
+                block_document=BlockDocumentCreate(
+                    name=credentials_block_name,
+                    data={
+                        "client_id": client_id,
+                        "tenant_id": tenant_id,
+                        "client_secret": client_secret,
+                    },
+                    block_type_id=credentials_block_type.id,
+                    block_schema_id=credentials_block_schema.id,
+                )
+            )
+
+            self._console.print(
+                (
+                    f"ACI credentials block '{credentials_block_name}' created in"
+                    " Prefect Cloud"
+                ),
+                style="green",
+            )
+            return block_doc.id
```

Comment:
Note to future us: We're doing this in at least 4 places now, so it's probably time to create a utility.